modules/share/src/godotShareData.mm

```diff
@@ -0,0 +1,100 @@
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "godotShareData.h"
+
+#import "app_delegate.h"
+
+GodotShareData::GodotShareData() {
+	ERR_FAIL_COND(instance != NULL);
+	instance = this;
+}
+
+GodotShareData::~GodotShareData() {
+	instance = NULL;
+}
+
+void GodotShareData::shareText(const String &title, const String &subject, const String &text) {
+	UIViewController *root_controller = [[UIApplication sharedApplication] delegate].window.rootViewController;
+
+	NSString *ns_text = [NSString stringWithCString:text.utf8().get_data() encoding:NSUTF8StringEncoding];
+	NSString *ns_subject = [NSString stringWithCString:subject.utf8().get_data() encoding:NSUTF8StringEncoding];
+
+	NSArray *shareItems = @[ ns_text ];
+
+	UIActivityViewController *avc = [[UIActivityViewController alloc] initWithActivityItems:shareItems applicationActivities:nil];
+	[avc setValue:ns_subject forKey:@"subject"];
+	//if iPhone
+	if (UI_USER_INTERFACE_IDIOM() == UIUserInterfaceIdiomPhone) {
+		[root_controller presentViewController:avc animated:YES completion:nil];
+	}
+	//if iPad
+	else {
+		// Change Rect to position Popover
+		avc.modalPresentationStyle = UIModalPresentationPopover;
+		avc.popoverPresentationController.sourceView = root_controller.view;
+		avc.popoverPresentationController.sourceRect = CGRectMake(CGRectGetMidX(root_controller.view.bounds), CGRectGetMidY(root_controller.view.bounds), 0, 0);
+		avc.popoverPresentationController.permittedArrowDirections = UIPopoverArrowDirection(0);
+		[root_controller presentViewController:avc animated:YES completion:nil];
+	}
+}
+
+void GodotShareData::shareImage(const String &path, const String &title, const String &subject, const String &text) {
```

Comment:
Not sure if it's useful with `path` as an argument (won't be able to access anything in the PCK, for example). This probably should take `Ref<Image>` instead and convert it to the `UImage` directly.

Reply:
Same API for android, internally in kotlin it doing
```
        val f = File(path)
        val uri: Uri = try {
            FileProvider.getUriForFile(activity!!, activity!!.packageName+fileProvider, f)
        } catch (e: IllegalArgumentException) {
            Log.e(logTag, "The selected file can't be shared: $path", e)
            return
        }
```
String-path just easier to cross"provide"